lib/contentProfileListener.js

```diff
@@ -165,6 +185,29 @@ class ContentProfileImporter extends SHRContentProfileParserListener {
     this._specs.contentProfiles.add(this._currentDef);
     this._currentDef = null;
   }
+  
+  // NOTE: This function "borrowed" from shr-expand
+  getRecursiveBasedOns(identifier, alreadyProcessed = []) {
+    // If it's primitive or we've already processed this one, don't go further (avoid circular dependencies)
+    if (identifier.isPrimitive || alreadyProcessed.some(id => id.equals(identifier))) {
```

Comment:
So... it strikes me that if the identifier passed in is a primitive, then it isn't returned in the result.  E.g., `getRecursiveBasedOns(new PrimitiveIdentifier('string'))` will return `[]`;  For non-primitives, however, it always returns at least a single-element array containing itself (thanks to line 203 below).

I think that should be fixed.

Reply:
I didn't need to edit the `getRecursiveBasedOns` function but I did add handling to support primitives.